src/App.jsx

```diff
@@ -1,10 +1,121 @@
+    const [countryInfo, setCountryInfo] = useState({});
+    const [error, setError] = useState('');
+
+    async function fetchCountries() {
+        try {
+            const response = await axios.get('https://restcountries.com/v3.1/all');
+            console.log(response.data);
+
+            response.data.sort((a, b) => {
+                return a.population - b.population;
+            });
+
+            setCountries(response.data);
+
+        } catch (e) {
+            console.error(e);
+        }
+    }
+
+    async function handleSubmit(event) {
+        event.preventDefault();
+        setError('');
+
+        try {
+            const response = await axios.get(`https://restcountries.com/v3.1/name/${searchCountry}`);
+            const country = response.data[0];
+            setCountryInfo(country);
+            setSearchCountry('');
+        } catch (e) {
+            console.error(e);
+            setError(`${searchCountry} bestaat niet. Probeer het opnieuw.`);
+        }
+    }
+
+    async function getCountry(getCountry){
+        console.log(getCountry);
+        setError('');
+        try {
+            const response = await axios.get(`https://restcountries.com/v3.1/name/${getCountry}`);
+            const country = response.data[0];
+            console.log(countryInfo);
+            setCountryInfo(country);
+            const element = document.querySelector(`.search-form`);
+            if (element) {
+                element.scrollIntoView({ behavior: 'smooth', block: 'start' });
+            }
+        } catch (e) {
+            console.error(e);
+            setError(`${getCountry} bestaat niet. Probeer het opnieuw.`);
+        }
+    }
+
     return (
         <>
-            Maak je applicatie hier!
+            <header>
+                <img src={map} alt="Wereldkaart" className="map"/>
+            </header>
+            <main>
+                <section className="inner-container">
+                    <h2>World Regions</h2>
+                    {countries.length > 0 ?
+                        <>
+                            <h2>Search country information</h2>
+                            <form className="search-form" onSubmit={handleSubmit}>
+                                <input
+                                    type="text"
+                                    name="query"
+                                    id="query-field"
+                                    placeholder="Bijvoorbeeld Nederland of Peru"
+                                    value={searchCountry}
+                                    onChange={(event) => setSearchCountry(event.target.value)}
+                                />
+                                <button type="submit">Zoek</button>
+                                {error && <span id="error-message">{searchCountry} bestaat niet. Probeer het opnieuw</span>}
+                            </form>
+                            {Object.keys(countryInfo).length > 0 &&
```

Comment:
Als je default waarde undefined is, dan kan het hier dus wat simpeler
```suggestion
                            {countryInfo &&
```

Reply:
Bedankt voor de feedback.